hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientExecutionServiceImpl.java

```diff
@@ -72,6 +72,8 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
                         throw new RejectedExecutionException(message);
                     }
                 });
+        ((LoggingScheduledExecutor) internalExecutor).enablePeriodicPurge();
```

Comment:
Not sure whether this is needed on client side or not. But I guess, when there are just a few scheduled tasks, this should be harmless.